### PR TITLE
[MRG] Allow search=auto in gp_minimize

### PR DESCRIPTION
**Pull request by MechCoder**
Clearly, lbfgs does not make sense for Integer and Categorical spaces since even though the acquisition function may be continuous, the actual function itself isn't. The lbfgs optimization procedure might lead to values in the transformed space in between 0 and 1, which might break inverse transform.

In this PR, search is set to lbfgs automatically only when all the dimensions are continuous.

### CODE REVIEW

**Comment by MechCoder** [comment]
Ok, this seems a bug in `inverse_transform` of `LabelBinarizer` in sklearn which is why it does not break.

```python
r = ["m", "a", "n", "o", "j"]
lb = LabelBinarizer()
lb.fit_transform(r)
t = np.asarray([[0.9, 0.1, 0, 0, 0]])
lb.inverse_transform(t)
# ???
array(['a'], 
      dtype='<U1')
```

**Comment by MechCoder** [comment]
Even for an `Integer` space, we wouldn't have the desired behaviour for `inverse_transform` because of casting to int. We should round it to the nearest integer.

File: skopt/space.py [hunk 1]
```diff
@@ -324,6 +324,13 @@ def __init__(self, dimensions):
 
         self.dimensions = _dimensions
 
+    @property
+    def is_real_space(self):
```
Comment (glouppe) [inline skopt/space.py]: `is_real` should be enough? (since typically this property will be checked through something like `space.is_real`)

**Comment by glouppe** [comment]
+1 once my small comment is addressed

**Comment by glouppe** [comment]
Set to `n_points=1000` as part of #174. Closing.

**Pull request merged**